src/WidgetBase.ts

```diff
@@ -271,17 +285,42 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			delete this._previousProperties[propertyName];
 		});
 
-		const diffPropertiesResult = this.diffProperties(this._previousProperties, properties);
-		this._properties = assign(diffPropertiesResult.properties, diffPropertyResults);
+		const diffPropertiesDecorators = this.getDecorator('diffProperties');
+		if (diffPropertiesDecorators.length) {
+			const diffDecorator = diffPropertiesDecorators[diffPropertiesDecorators.length - 1];
```

Comment:
Yes I think that makes the most sense, in the extended widget if you need a different behaviour for a property then as you say you can use `diffProperty`.

It does raise the question about diffing for a specific property, can they be overridden? Are they overridden? or will both diffs run?